cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/analyzer/goals/CapacityGoal.java

```diff
@@ -188,6 +188,8 @@ protected void updateGoalState(ClusterModel clusterModel, Set<String> excludedTo
     ensureUtilizationUnderCapacity(clusterModel);
     // Sanity check: No self-healing eligible replica should remain at a dead broker/disk.
     AnalyzerUtils.ensureNoOfflineReplicas(clusterModel);
+    // Sanity check: No replica should be move to broker which used to host the replica of same partition on its broken disk.
```

Comment:
Nit: `// Sanity check: No replica should be move to broker which used to host the replica of same partition on its broken disk.` -> `// Sanity check: No replica should be moved to a broker, which used to host any replica of the same partition on its broken disk.`